src/devices/Common/Iot/Device/Multiplexing/IOutputSegment.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Device.Gpio;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Iot.Device.Multiplexing
+{
+    /// <summary>
+    /// Abstracts a segment of outputs from multiplexing sources (like a shift register).
+    /// </summary>
+    public interface IOutputSegment : IDisposable
+    {
+        /// <summary>
+        /// Length of segment (number of outputs)
+        /// </summary>
+        int Length { get; }
+
+        /// <summary>
+        /// Segment values.
+        /// </summary>
+        PinValue this[int index] { get; set; }
+
+        /// <summary>
+        /// Writes a PinValue to a virtual segment.
+        /// Does not display output.
```

Comment:
One more small nit here, perhaps say

```suggestion
        /// Does not display output until calling Display() or Display(CancellationToken ct) methods.
```